src/components/controllers/slotted-controller.ts

```diff
@@ -19,19 +23,40 @@ export class SlottedController implements ReactiveController {
   }
 
   onSlotChange() {
-    const elementName = this.hostEl.tagName;
-    const myElement: HTMLElement | null = this.hostEl.parentElement
-      ? this.hostEl.parentElement.querySelector(elementName)
-      : null;
-    if (myElement && myElement.shadowRoot) {
-      const slot: HTMLElement | null = myElement.shadowRoot.querySelector(
-        'slot'
-      )
-        ? myElement.shadowRoot.querySelector('slot')
-        : null;
-      if (slot) {
-        slot.append(...myElement.children);
-      }
+    // Search for all slots provided by the Web Component.
+    const slotsArray = this.hostEl.renderRoot.querySelectorAll('slot');
+
+    // If any slots found
+    if (slotsArray.length > 0) {
+      const currentElement = this.hostEl;
+      // Go through each slot shadow dom position
+      slotsArray.forEach(slot => {
+        // Process named slot
+        if (slot.name) {
+          // Fetch the corresponding named slot in the lightDom
+          const slotLightDom = currentElement.querySelector(
+            '[slot=' + slot.name + ']'
+          );
+          // Only if named slot found in light dom - move it into shadow DOM
+          if (slotLightDom) {
+            slot.before(slotLightDom);
+          }
+        }
+        // Process default (unnamed) slot
+        else {
+          // Get all content that doesn't have slot as attribute
+          const slotLightDomArray = Array.from(currentElement.children).filter(
+            node => {
+              return !node.getAttributeNode('slot');
+            }
+          );
+
+          // Move all unnamed slot content to the corresponding position in shadow DOM
+          slotLightDomArray.forEach(slotLightDom => {
+            slot.before(slotLightDom);
+          });
+        }
+      });
     }
```

Comment:
return early for easier to read code and less nesting.

```suggestion
    // Nothing to do if we have no slots.
    if (!slotsArray.length) {
      return;
    }
    
    const currentElement = this.hostEl;
    // Go through each slot shadow dom position
    slotsArray.forEach(slot => {
      // Process named slot
      if (slot.name) {
        // Fetch the corresponding named slot in the lightDom
        const slotLightDom = currentElement.querySelector(
          '[slot=' + slot.name + ']'
        );
        // Only if named slot found in light dom - move it into shadow DOM
        if (slotLightDom) {
          slot.before(slotLightDom);
        }
      }
      // Process default (unnamed) slot
      else {
        // Get all content that doesn't have slot as attribute
        const slotLightDomArray = Array.from(currentElement.children).filter(
          node => {
            return !node.getAttributeNode('slot');
          }
        );

        // Move all unnamed slot content to the corresponding position in shadow DOM
        slotLightDomArray.forEach(slotLightDom => {
          slot.before(slotLightDom);
        });
      }
    });
```